pkgs/applications/terminal-emulators/rio/default.nix

```diff
@@ -1,57 +1,65 @@
-{ lib
-, stdenv
-, darwin
-, fetchFromGitHub
-, rustPlatform
-, nixosTests
-, nix-update-script
-
-, autoPatchelfHook
-, cmake
-, ncurses
-, pkg-config
-
-, gcc-unwrapped
-, fontconfig
-, libGL
-, vulkan-loader
-, libxkbcommon
-
-, withX11 ? !stdenv.hostPlatform.isDarwin
-, libX11
-, libXcursor
-, libXi
-, libXrandr
-, libxcb
-
-, withWayland ? !stdenv.hostPlatform.isDarwin
-, wayland
-
-, testers
-, rio
+{
+  lib,
+  stdenv,
+  darwin,
+  fetchFromGitHub,
+  rustPlatform,
+  nixosTests,
+  nix-update-script,
+
+  autoPatchelfHook,
+  cmake,
+  ncurses,
+  pkg-config,
+
+  gcc-unwrapped,
+  fontconfig,
+  libGL,
+  vulkan-loader,
+  libxkbcommon,
+  apple-sdk_11,
+
+  withX11 ? !stdenv.hostPlatform.isDarwin,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  libxcb,
+
+  withWayland ? !stdenv.hostPlatform.isDarwin,
+  wayland,
+
+  testers,
+  rio,
 }:
 let
-  rlinkLibs = if stdenv.hostPlatform.isDarwin then [
-    darwin.libobjc
-    darwin.apple_sdk_11_0.frameworks.AppKit
-    darwin.apple_sdk_11_0.frameworks.AVFoundation
-    darwin.apple_sdk_11_0.frameworks.MetalKit
-    darwin.apple_sdk_11_0.frameworks.Vision
-  ] else [
-    (lib.getLib gcc-unwrapped)
-    fontconfig
-    libGL
-    libxkbcommon
-    vulkan-loader
-  ] ++ lib.optionals withX11 [
-    libX11
-    libXcursor
-    libXi
-    libXrandr
-    libxcb
-  ] ++ lib.optionals withWayland [
-    wayland
-  ];
+  rlinkLibs =
+    if stdenv.hostPlatform.isDarwin then
+      [
+        darwin.libobjc
+        darwin.apple_sdk_11_0.frameworks.AppKit
+        darwin.apple_sdk_11_0.frameworks.AVFoundation
+        darwin.apple_sdk_11_0.frameworks.MetalKit
+        darwin.apple_sdk_11_0.frameworks.Vision
+      ]
```

Comment:
Are these still needed?

Reply:
You are right, these can be removed. I built and tested both architectures without these and seems to work fine. I pushed an update to my commits.